doc/CMakeLists.txt

```diff
@@ -4,47 +4,57 @@ ELSE ( ENABLE_BUILD_DOCS )
   ADD_SUBDIRECTORY( autodoc EXCLUDE_FROM_ALL )
 ENDIF ( ENABLE_BUILD_DOCS )
 
+FIND_PROGRAM(A2X a2x)
+IF (NOT A2X)
+MESSAGE(FATAL_ERROR "Could not find a2x, please install asciidoc")
+ELSE (NOT A2X)
+MESSAGE("a2x found: ${A2X}")
+ENDIF (NOT A2X)
+
+function(ADD_A2X_MANPAGES)
+    set(options)
+    set(oneValueArgs MAN_INST_PATH)
+    set(multiValueArgs MAN_PAGES1 MAN_PAGES2 MAN_PAGES3 MAN_PAGES4 MAN_PAGES5 MAN_PAGES6 MAN_PAGES7 MAN_PAGES8)
+    cmake_parse_arguments(A2X "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+    set(ALL_MAN_PAGES)
+
+    FOREACH (MAN_IDX RANGE 1 8)
+        IF (NOT DEFINED A2X_MAN_PAGES${MAN_IDX})
+            CONTINUE()
+        ENDIF()
+
+        FOREACH (curr_PAGE ${A2X_MAN_PAGES${MAN_IDX}})
+            ADD_CUSTOM_COMMAND (
+              OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${curr_PAGE}
+              COMMAND ${A2X} -D ${CMAKE_CURRENT_BINARY_DIR} -d manpage -f manpage ${CMAKE_CURRENT_SOURCE_DIR}/${curr_PAGE}.txt
```

Comment:
cd /Local/ma/zypp/BUILD/libzypp/doc && \
        /usr/bin/a2x -D /Local/ma/zypp/BUILD/libzypp/doc -d manpage -f  manpage /Local/ma/zypp/libzypp/doc/locks.5.txt
    a2x: WARNING: --destination-dir option is only applicable to HTML based outputs

So `-D` isn't supported. As cmake has `WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}` it nevertheless works. As I understand the 2ax manpage `-f manpage` also implies `-d manpage`. Both switches should neot be needed.

Reply:
There is a bug in asciidoc about this being misleading because -D is implemented for manpage as well

Reply:
Done

Reply:
https://github.com/asciidoc/asciidoc/issues/44